### PR TITLE
net-irc/inspircd-2.0.23: Version bump (#593290)

### PR DESCRIPTION
**Pull request by awilfox**
Package-Manager: portage-2.3.1

### CODE REVIEW

File: net-irc/inspircd/inspircd-2.0.23.ebuild [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	use ldap && extras="${extras}m_ldapauth.cpp,m_ldapoper.cpp,"
+	use mysql && extras="${extras}m_mysql.cpp,"
+	use pcre && extras="${extras}m_regex_pcre.cpp,"
+	use posix && extras="${extras}m_regex_posix.cpp,"
+	use postgres && extras="${extras}m_pgsql.cpp,"
+	use sqlite && extras="${extras}m_sqlite3.cpp,"
+	use ssl && extras="${extras}m_ssl_openssl.cpp,"
+	use tre && extras="${extras}m_regex_tre.cpp,"
+
+	if [ -n "${extras}" ]; then
```
Comment (SoapGentoo) [inline net-irc/inspircd/inspircd-2.0.23.ebuild]: always use `[[ ]]` parentheses

File: net-irc/inspircd/inspircd-2.0.23.ebuild [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+src_compile() {
+	emake V=1 LDFLAGS="${LDFLAGS}" CXXFLAGS="${CXXFLAGS}"
+}
+
+src_install() {
+	emake INSTUID=${PN} DESTDIR="${D}" install
+
+	insinto "/usr/include/${PN}"
+	doins include/*
```
Comment (SoapGentoo) [inline net-irc/inspircd/inspircd-2.0.23.ebuild]: try `doins -r include/.`

File: net-irc/inspircd/inspircd-2.0.23.ebuild [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~ppc64 ~x86"
+IUSE="geoip gnutls ipv6 ldap mysql pcre posix postgres sqlite ssl tre"
+
+RDEPEND="
+	dev-lang/perl
+	ssl? ( dev-libs/openssl:= )
+	geoip? ( dev-libs/geoip )
+	gnutls? ( net-libs/gnutls dev-libs/libgcrypt:0 )
```
Comment (SoapGentoo) [inline net-irc/inspircd/inspircd-2.0.23.ebuild]: gnutls and libcrypt normally need `:=` and `:0=` respectively
also, are gnutls and openssl support simultaneously?

File: net-irc/inspircd/inspircd-2.0.23.ebuild [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+SRC_URI="https://github.com/inspircd/inspircd/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~ppc64 ~x86"
+IUSE="geoip gnutls ipv6 ldap mysql pcre posix postgres sqlite ssl tre"
+
+RDEPEND="
+	dev-lang/perl
+	ssl? ( dev-libs/openssl:= )
```
Comment (SoapGentoo) [inline net-irc/inspircd/inspircd-2.0.23.ebuild]: `:0=`

File: net-irc/inspircd/inspircd-2.0.23.ebuild (hunk 3 above)
Comment (awilfox) [inline net-irc/inspircd/inspircd-2.0.23.ebuild]: They most certainly are supported at the same time.  I have it built that way locally, even.  Some features need OpenSSL, while you can still link servers together using GnuTLS.

**Comment by awilfox** [comment]
I am a little surprised at the pedantry over a commit that provides a long-needed security fix for a severe CVE.

It is late at night here, so this will have to wait until I have more time to deal with fussing with Git.  In the future I won't be making any PRs, because the workflow is too much of a disaster.  I will just use Bugzilla where it is easy to replace an attachment.

File: net-irc/inspircd/inspircd-2.0.23.ebuild (hunk 3 above)
Comment (SoapGentoo) [inline net-irc/inspircd/inspircd-2.0.23.ebuild]: perfect then

**Comment by Whissi** [comment]
@awilfox: I am going to commit your changes with the following adjustments. Please tell me if you are OK with my changes and want to retain authorship or if I should commit using my own name or if you disagree with the changes in general:
```diff
diff -purN /tmp/awilfox-gentoo/net-irc/inspircd/files/inspircd-2.0.23-init /usr/portage/net-irc/inspircd/files/inspircd-2.0.23-init
--- /tmp/awilfox-gentoo/net-irc/inspircd/files/inspircd-2.0.23-init	2017-02-21 16:59:04.638980278 +0100
+++ /usr/portage/net-irc/inspircd/files/inspircd-2.0.23-init		1970-01-01 01:00:00.000000000 +0100
@@ -1,39 +0,0 @@
-#!/sbin/openrc-run
-# Copyright 1999-2017 Gentoo Foundation
-# Distributed under the terms of the GNU General Public License v2
-# $Id$
-
-extra_commands="rehash version"
-
-depend() {
-	need net
-	provide ircd
-}
-
-start() {
-	ebegin "Starting InspIRCd"
-	checkpath -d -m 0750 -o inspircd:inspircd /var/run/inspircd
-	start-stop-daemon --start --quiet --user inspircd \
-		--exec /usr/bin/inspircd -- \
-		--config /etc/inspircd/inspircd.conf \
-		--logfile /var/log/inspircd/ircd.log
-	eend $?
-}
-
-stop() {
-	ebegin "Stopping InspIRCd"
-	start-stop-daemon --stop --quiet --exec /usr/bin/inspircd
-	eend $?
-}
-
-rehash() {
-	ebegin "Rehashing InspIRCd"
-	/usr/lib/inspircd/inspircd rehash
-	eend $?
-}
-
-version() {
-	ebegin "Retrieve InspIRCd version"
-	/usr/lib/inspircd/inspircd version
-	eend $?
-}
diff -purN /tmp/awilfox-gentoo/net-irc/inspircd/files/inspircd.confd /usr/portage/net-irc/inspircd/files/inspircd.confd
--- /tmp/awilfox-gentoo/net-irc/inspircd/files/inspircd.confd	1970-01-01 01:00:00.000000000 +0100
+++ /usr/portage/net-irc/inspircd/files/inspircd.confd		2017-02-21 16:56:29.975850074 +0100
@@ -0,0 +1,39 @@
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+# Location of configuration file. Modify if you don't like the standard
+# one.
+#
+#INSPIRCD_CONFIGFILE="/etc/inspircd/inspircd.conf"
+
+# Location of log file. Modify if you don't like the standard
+# one.
+#
+#INSPIRCD_LOGFILE="/var/log/inspircd/ircd.log"
+
+# File used to store the PID file. Usually you won't need to touch it.
+# If you are changing PID file do not forget to set new PID file in
+# INSPIRCD_CONFIGFILE as well.
+#
+#INSPIRCD_PIDFILE="/run/inspircd/inspircd.pid"
+
+# User to run InspIRCd as
+#
+#INSPIRCD_USER="inspircd"
+#INSPIRCD_GROUP="inspircd"
+
+# You can use this configuration option to pass additional options to the
+# start-stop-daemon, see start-stop-daemon(8) for more details.
+# Per default we start InspIRCd without any output (quiet)
+# as user $INSPIRCD_USER and wait 1000ms after we have started the service
+# to ensure that the daemon is really up and running.
+#INSPIRCD_SSDARGS="--quiet --user \"${INSPIRCD_USER}\" --wait 1000"
+
+# The termination timeout (start-stop-daemon parameter "retry") ensures
+# that the service will be terminated within a given time (25 + 5 seconds
+# per default) when you are stopping the service.
+#INSPIRCD_TERMTIMEOUT="TERM/25/KILL/5"
+
+# Options to collectd
+# See `/usr/bin/inspircd --help` for more details
+#INSPIRCD_OPTS=""
diff -purN /tmp/awilfox-gentoo/net-irc/inspircd/files/inspircd-r2.initd /usr/portage/net-irc/inspircd/files/inspircd-r2.initd
--- /tmp/awilfox-gentoo/net-irc/inspircd/files/inspircd-r2.initd	1970-01-01 01:00:00.000000000 +0100
+++ /usr/portage/net-irc/inspircd/files/inspircd-r2.initd		2017-02-21 16:54:31.414447855 +0100
@@ -0,0 +1,49 @@
+#!/sbin/openrc-run
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+INSPIRCD_USER=${INSPIRCD_USER:-"inspircd"}
+INSPIRCD_GROUP=${INSPIRCD_GROUP:-"inspircd"}
+INSPIRCD_CONFIGFILE=${INSPIRCD_CONFIGFILE:-"/etc/inspircd/inspircd.conf"}
+INSPIRCD_LOGFILE=${INSPIRCD_LOGFILE:-"/var/log/inspircd/ircd.log"}
+INSPIRCD_PIDFILE=${INSPIRCD_PIDFILE:-"/run/inspircd/inspircd.pid"}
+INSPIRCD_SSDARGS=${INSPIRCD_SSDARGS:-"--quiet --user \"${INSPIRCD_USER}\" --wait 1000"}
+INSPIRCD_TERMTIMEOUT=${INSPIRCD_TERMTIMEOUT:-"TERM/25/KILL/5"}
+INSPIRCD_OPTS=${INSPIRCD_OPTS:-""}
+
+command="/usr/bin/inspircd"
+command_args="${INSPIRCD_OPTS} --config \"${INSPIRCD_CONFIGFILE}\" --logfile \"${INSPIRCD_LOGFILE}\""
+start_stop_daemon_args="${INSPIRCD_SSDARGS}"
+pidfile="${INSPIRCD_PIDFILE}"
+retry="${INSPIRCD_TERMTIMEOUT}"
+
+required_files="${INSPIRCD_CONFIGFILE}"
+
+name="InspIRCd"
+
+description="InspIRCd is a modular Internet Relay Chat (IRC) server."
+
+extra_commands="version"
+extra_started_commands="rehash"
+
+depend() {
+	use dns
+	provide ircd
+}
+
+start_pre() {
+	checkpath --directory --mode 0750 --owner ${INSPIRCD_USER}:${INSPIRCD_GROUP} "$(dirname "${INSPIRCD_PIDFILE}")"
+}
+
+rehash() {
+	ebegin "Rehashing InspIRCd"
+	start-stop-daemon --signal HUP --pidfile "${pidfile}"
+	eend $?
+}
+
+version() {
+	ebegin "Retrieve InspIRCd version"
+	${command} --version
+	eend $?
+}
diff -purN /tmp/awilfox-gentoo/net-irc/inspircd/inspircd-2.0.23.ebuild /usr/portage/net-irc/inspircd/inspircd-2.0.23.ebuild
--- /tmp/awilfox-gentoo/net-irc/inspircd/inspircd-2.0.23.ebuild	2017-02-21 16:59:04.638980278 +0100
+++ /usr/portage/net-irc/inspircd/inspircd-2.0.23.ebuild	2017-02-21 16:49:45.307674431 +0100
@@ -2,7 +2,7 @@
 # Distributed under the terms of the GNU General Public License v2
 # $Id$
 
-EAPI=6
+EAPI="6"
 
 inherit toolchain-funcs user
 
@@ -19,7 +19,7 @@ RDEPEND="
 	dev-lang/perl
 	ssl? ( dev-libs/openssl:= )
 	geoip? ( dev-libs/geoip )
-	gnutls? ( net-libs/gnutls dev-libs/libgcrypt:0 )
+	gnutls? ( net-libs/gnutls:= dev-libs/libgcrypt:0 )
 	ldap? ( net-nds/openldap )
 	mysql? ( virtual/mysql )
 	postgres? ( dev-db/postgresql:= )
@@ -56,8 +56,8 @@ src_configure() {
 	use ssl && extras="${extras}m_ssl_openssl.cpp,"
 	use tre && extras="${extras}m_regex_tre.cpp,"
 
-	if [ -n "${extras}" ]; then
-		econf --disable-interactive --enable-extras=${extras}
+	if [[ -n "${extras}" ]]; then
+		econf --disable-interactive --enable-extras=${extras%,}
 	fi
 
 	econf \
@@ -69,9 +69,9 @@ src_configure() {
 		--log-dir="/var/log/${PN}" \
 		--binary-dir="/usr/bin" \
 		--module-dir="/usr/$(get_libdir)/${PN}/modules" \
-		$(use_enable ipv6) \
-		$(use_enable gnutls) \
-		$(use_enable ssl openssl)
+		$(usex ipv6 '' '--disable-ipv6') \
+		$(usex gnutls '--enable-gnutls' '') \
+		$(usex ssl '--enable-openssl' '')
 }
 
 src_compile() {
@@ -79,24 +79,29 @@ src_compile() {
 }
 
 src_install() {
-	emake INSTUID=${PN} DESTDIR="${D}" install
+	emake INSTUID=${PN} DESTDIR="${D%/}" install
 
 	insinto "/usr/include/${PN}"
-	doins include/*
+	doins -r include/.
 
 	diropts -o"${PN}" -g"${PN}" -m0700
 	dodir "/var/lib/${PN}"
 	dodir "/var/lib/${PN}/data"
 
-	newinitd "${FILESDIR}/${P}-init" "${PN}"
-	keepdir "/var/log/${PN}"/
+	newinitd "${FILESDIR}/${PN}-r2.initd" "${PN}"
+	newconfd "${FILESDIR}/${PN}.confd" "${PN}"
+
+	keepdir "/var/log/${PN}"
 }
 
 pkg_postinst() {
-	elog "Before starting ${PN} the first time, you should create"
-	elog "the /etc/${PN}/${PN}.conf file."
-	elog "You can find example configuration files under /etc/${PN}"
-	elog "Read the ${PN}.conf.example file carefully before "
-	elog "(re)starting the service."
-	elog
+	if [[ -z "${REPLACING_VERSIONS}" ]]; then
+		# This is a new installation
+		elog "Before starting ${PN} the first time, you should create"
+		elog "the /etc/${PN}/${PN}.conf file."
+		elog "You can find example configuration files under /etc/${PN}"
+		elog "Read the ${PN}.conf.example file carefully before "
+		elog "(re)starting the service."
+		elog
+	fi
 }
```

What I changed:

- I addressed outstanding issues reported by @SoapGentoo 
- Added some variable substitution (removed "," from last *extras* value, removed double slash from $D in *src_install*)
- Options like if IPv6 support are one-way only, i.e. you will get a non-fatal error like "Unknown option" if you explicitly try to enable IPv6 support; I used *usex* instead to only set a value if needed
- I added a missing slot operator on *net-libs/gnutls*
- *elog* in *pkg_postinst* will now only be shown on first install
- I have rewritten/updated runscript to add some love; In theory you could now run multiple instances, we use default start/stop functions and now have support for OpenRC's built-in status function
- Renamed runscripts so that we don't need to adjust them on bump

**Comment by Whissi** [comment]
- Runscript no longer need net, see https://bugs.gentoo.org/show_bug.cgi?id=439092

**Comment by awilfox** [comment]
I approve your changes @Whissi and you may commit them with my authorship.  Thank you!

**Pull request closed without merging**